src/renderer/store/modules/utils.js

```diff
@@ -888,6 +901,146 @@ const mutations = {
     state.isKeyboardShortcutPromptShown = payload
   },
 
+  setAreVimWaypointsShown(state, { key }) {
+    const selectedLetterColor = '#D4AC3A'
+    if (key === 'w' && !document.querySelector('.vimHint')) {
+      const elements = [
+        ...(document.querySelector('.prompt') || document).querySelectorAll('a, button, [type="button"], [type="submit"], [role="tab"], [role="button"], input[type="text"], [role="link"]')
+      ]
+
+      if (!elements.length) {
+        state.areVimWaypointsShown.selector = []
+        return
+      }
+      let hintIndex = 0
+
+      const minChars = 2
+      const charset = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+      const hints = generateHints(minChars, charset)
+
+      elements.forEach((element) => {
+        const containerSpan = document.createElement('span')
+        const hintStyles = {
+          position: 'absolute',
+          background: 'linear-gradient(to bottom, #fff204 0%, #d8cb03 100%)',
+          fontSize: '13px',
+          fontFamily: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Arial, sans-serif',
+          fontWeight: '600',
+          borderRadius: '3px',
+          color: 'black',
+          zIndex: '10000', // Ensure it is above other content
+          padding: '1px 2px',
+          letterSpacing: '0.5px',
+          display: 'flex' // Use flexbox for letter alignment
+        }
```

Comment:
As you are already assigning a class to this element, it would make a lot more sense to put the CSS in the App.css file. You'll need to use the `:deep()` selector as you are creating these elements outside of Vue.